components/library/tntdb/Makefile

```diff
@@ -0,0 +1,61 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+# BuildRequires: cxxtools, sqlite-3, pg_config (postgres-9*/developer),
+# mysql_config (mariadb-55/client, percona, mysql)
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=	tntdb
+COMPONENT_VERSION=	1.3
+IPS_COMPONENT_VERSION=	1.3.0
+COMPONENT_SUMMARY=	C++ library to access SQL databases, easy and robust
+COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE= $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH= \
+  sha256:093ab2694bc66bc05ae0a5c8a505d75ae84d4c35c2ddc0ff0283c5b9574b474b
+COMPONENT_ARCHIVE_URL= \
+  http://www.tntnet.org/download/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL=	http://www.tntnet.org/tntdb.html
+COMPONENT_FMRI= library/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION= System/Libraries
+COMPONENT_LICENSE = LGPLv2.1
+COMPONENT_LICENSE_FILE = COPYING
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+# Sorry, this listing of our possible mysql implementations (in this repo)
```

Comment:
An OI-Hipster user can also install some other postgresql version (we have
components for 9.3, 9.4 and 9.5 at least), as well as another mysql version
or fork (percona is among them, IIRC?) The actual settings for the build
are taken from configure script's processing of "mysql_config" and
"pg_config" as found in PATH, and for the variants we have in userland
these are mediated somehow.

For the binary API, it "should not" matter which one a consumer is linked
against, and a customer of e.g. percona might be surprised about pulling
mariadb instead to fulfill this package requirements.

Although likely we should then ensure linking against the oldest version
available, or otherwise least common denominator (to only use ubiquitously
available versioned symbols, etc.) - so my theoretical opinion above may be
not that firm in practice.

2016-03-21 19:05 GMT+01:00 Alexander Pyhalov <notifications@github.com>:

> In components/library/tntdb/Makefile
> <https://github.com/OpenIndiana/oi-userland/pull/1776#discussion_r56868285>
> :
>
> > +COMPONENT_ARCHIVE= $(COMPONENT_SRC).tar.gz
> > +COMPONENT_ARCHIVE_HASH= \
> > +  sha256:093ab2694bc66bc05ae0a5c8a505d75ae84d4c35c2ddc0ff0283c5b9574b474b
> > +COMPONENT_ARCHIVE_URL= \
> > +  http://www.tntnet.org/download/$(COMPONENT_ARCHIVE)
> > +COMPONENT_PROJECT_URL=	http://www.tntnet.org/tntdb.html
> > +COMPONENT_FMRI= library/$(COMPONENT_NAME)
> > +COMPONENT_CLASSIFICATION= System/Libraries
> > +COMPONENT_LICENSE = LGPLv2.1
> > +COMPONENT_LICENSE_FILE = COPYING
> > +
> > +include $(WS_TOP)/make-rules/prep.mk
> > +include $(WS_TOP)/make-rules/configure.mk
> > +include $(WS_TOP)/make-rules/ips.mk
> > +
> > +# Sorry, this listing of our possible mysql implementations (in this repo)
>
> We use mariadb 5.5 as default mysql implementation and PG_VERSION as
> default postgresql version. Please, fix PATH.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/1776/files/850a36867657f82c6ac7c3603e69250f1299896a#r56868285>
>